examples/siro_sandbox/test_episode_save_files.py

```diff
@@ -0,0 +1,111 @@
+#!/usr/bin/env python3
+
+# Copyright (c) Meta Platforms, Inc. and its affiliates.
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+import argparse
+
+import numpy as np
+from serialize_utils import load_json_gzip, load_pickle_gzip
+
+
+def get_comp_value(a):
+    if isinstance(a, np.generic):
+        # If obj is a NumPy scalar type, convert it to a standard Python scalar type
+        return a.item()
+    else:
+        return a
+
+
+# Not completely general-purpose. Works for simple types and list-like types.
+def approx_equal(a_src, b_src, eps=1e-4):
+    a = get_comp_value(a_src)
+    b = get_comp_value(b_src)
+
+    if isinstance(a, (int, str, bool, type(None))):
+        return a == b
+    elif isinstance(a, float):
+        return abs(a - b) < eps
+    else:
+        a_list = a
+        b_list = b
+        if len(a_list) != len(b_list):
+            return False
+
+        return all(
+            approx_equal(a_list[i], b_list[i]) for i in range(len(a_list))
+        )
+
+
+def test_episode_save_files(filepath_base):
+    json_filepath = filepath_base + ".json.gz"
+    pkl_filepath = filepath_base + ".pkl.gz"
+
+    pkl_obj = load_pickle_gzip(pkl_filepath)
+    json_obj = load_json_gzip(json_filepath)
+
+    # Spot-check selected items for equality. We don't expect equality in all items
+    # because of differences in how the two data files are serialized/deserialized.
+    # For example, for json, some complex types are converted to strings and floats
+    # are rounded to 5 digits of precision.
+
+    for key in ["episode_id", "target_obj_ids"]:
+        assert pkl_obj[key] == json_obj[key]
+
+    for key in ["goal_positions"]:
+        assert len(pkl_obj[key])
+        assert approx_equal(pkl_obj[key], json_obj[key])
+
+    for step_idx in range(len(pkl_obj["steps"])):
+        pkl_step = pkl_obj["steps"][step_idx]
+        json_step = json_obj["steps"][step_idx]
+
+        assert approx_equal(
+            pkl_step["gui_humanoid"]["cam_yaw"],
+            json_step["gui_humanoid"]["cam_yaw"],
+        )
+
+        pkl_subobj = pkl_step["action"]["action_args"]
+        json_subobj = json_step["action"]["action_args"]
+        keys = [
+            "agent_0_arm_action",
+            "agent_0_oracle_nav_with_backing_up_action",
```

Comment:
I'm a little concerned about our future ability to make sense of `agent_0_oracle_nav_with_backing_up_action`. It tends to be an index into the PDDL entity list. For example, I saw that it was `3` for navigating to target object #0. I'm not sure how we'll make sense of that 3 when consuming this data.

Reply:
Yeah, I agree. We reconstruct the PDDL to get what the corresponding action is. But we do also return the action executed as a string as well to the info dictionary [here](https://github.com/facebookresearch/habitat-lab/blob/5d3127746925fc232841c2183be795b29719a631/habitat-baselines/habitat_baselines/rl/hrl/hl/neural_policy.py#L240). So this should provide us all the context we need.